extension/asapclientauthextension/extension_test.go

```diff
@@ -0,0 +1,110 @@
+package asapclientauthextension
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"bitbucket.org/atlassian/go-asap"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// mockRoundTripper copies the request headers to the response.
+// So, the caller can make assertions using the returned response.
+type mockRoundTripper struct{}
+
+func (m *mockRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp := &http.Response{StatusCode: http.StatusOK, Header: map[string][]string{}}
+	for k, v := range req.Header {
+		resp.Header.Set(k, v[0])
+	}
+	return resp, nil
+}
+
+var _ http.RoundTripper = (*mockRoundTripper)(nil)
+
+// mockKeyFetcher implements asap.KeyFetcher, eliminating the need to contact a key server.
+type mockKeyFetcher struct{}
+
+func (k *mockKeyFetcher) Fetch(_ string) (interface{}, error) {
+	return asap.NewPublicKey([]byte(TestPubKey))
+}
+
+var _ asap.KeyFetcher = (*mockKeyFetcher)(nil)
+
+func TestRoundTripper(t *testing.T) {
+	cfg := &Config{
+		PrivateKey: TestPvtKey,
+		Ttl:        60,
+		Audience:   []string{"test"},
+		Issuer:     "test_issuer",
```

Comment:
Issuer and KeyID should be made constant in this file since they are set more than once

Reply:
I feel like this makes it slightly less readable for not much benefit

Reply:
Yeah, makes sense :) 

My thoughts behind this was to reduce potential typos in values however, if there is no benefit from this and the scope is small. I think it is safe to leave as is.